### PR TITLE
Replace npm with yarn to install/run the webui

### PR DESCRIPTION
**Pull request by tomMoulard**
### What does this PR do?

This PR changes the way that Web UI dependencies are installed through docker.
It now uses [`yarn install`](https://www.npmjs.com/package/yarn) instead of `npm install`.

This PR also replaces the usage of `npm run` by `yarn` to avoid using two utilities when only one is required.

### Motivation

This PR helps to build the Web UI locally by speeding up the process.

### More

- ~[ ] Added/updated tests~
- ~[ ] Added/updated documentation~

### Additional Notes

For my tests, I saw a decrease of building time, going from 201.5s to 32.2s.

I also tried to use [`pnpm`](https://pnpm.io), but it requires more changes on the building process.

This feature was already introduced in #846, but was revert in #5249.

### CODE REVIEW

**Review by rtribotte — APPROVED** [review]
LGTM 👌

**Review by 0xflotus — CHANGES_REQUESTED** [review]
You should consider to copy the resulting yarn.lock also to the docker image. 
https://github.com/traefik/traefik/blob/11ff7415121ad4231e7b95d984ab2670f672c371/webui/Dockerfile#L9

Now you copy the package-lock.json. If you use yarn, yarn will create a yarn.lock as equivalent.

File: webui/Dockerfile [hunk 1]
```diff
@@ -7,10 +7,11 @@ ENV PLATFORM_URL=${ARG_PLATFORM_URL}
 RUN mkdir -p $WEBUI_DIR
 
 COPY package.json $WEBUI_DIR/
-COPY package-lock.json $WEBUI_DIR/
+COPY yarn.json $WEBUI_DIR/
```
Comment (0xflotus) [inline webui/Dockerfile:10]: The file is called `yarn.lock`

**Review by 0xflotus — APPROVED** [review]

**Review by kevinpollet — CHANGES_REQUESTED** [review]
If we are switching to `yarn`, I would prefer to replace every `npm` command by the `yarn` equivalent to avoid using two different tools for the webui.

**Review by kevinpollet — APPROVED** [review]
LGTM 🚀

**Pull request merged**